### PR TITLE
Treat list custom json facebook

### LINKED ISSUE #5657: Type error in facebook.py
Hi,

I tried deploying the bot with Facebook and one error appears when I start a form:

TypeError: pop() takes at most 1 argument (2 given) (in rasa/core/channels/facebook.py : line 229).

I did a workaround but I thought you guys might want to take a look at this.

Cheers.

### PR DESCRIPTION
**Pull request by wavymazy**
**Context:**

I have retested the issue with Facebook to understand where it comes from.
The issue stems from the custom template/response if you have any. Basically, if we follow the template (https://rasa.com/docs/rasa/responses/#custom-output-payloads), it is said that we can send any arbitrary output we want. But what will end up happening if we have 2 "type" items (under "blocks"), is that the output will be a list (under "blocks"). The MessengerBot.send_custom_json() previously didn't handle this well at all.

Example of a custom Json that would cause issues:

```
{
        "blocks": [
            {"type": "title", "text": {"text": "Conversation progress"}},
            {
                "type": "progression_bar",
                "text": {"text": "progression 1", "level": "1"},
            },
        ]
    }
```

**Proposed changes**:
- Since the custom template can take any shape, this proposed change verifies that the custom json has a sender_id specified. Previously, it was assumed that the custom json is always a dictionary of dictionaries when it could be a dict containing a list for example, this would result in an error when using pop() as this method differs for dictionaries and lists).
- fixes #5657 
- Duplicate of #7247  

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by wavymazy** [comment]
@wochinge Here is the pull request I made (derived from https://github.com/RasaHQ/rasa/pull/7247). Should I add you as a reviewer ? And how ?

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @joejuzl will take a look at it as soon as possible ✨

File: tests/test_facebook.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            recipient_id,
+            messaging_type="RESPONSE",
+            notification_type="REGULAR",
+            timeout=None,
+            tag=None,
+        ):
+            self.recipient_id = recipient_id
+
+    messenger_client = TestableMessengerClient(page_access_token="test_token")
+    messenger_bot = MessengerBot(messenger_client)
```
Comment (wochinge) [inline tests/test_facebook.py:103]: what about using the `Mock` class from `unittest` instead of the custom class?

**Review by wochinge — COMMENTED** [review]
Thanks for wrapping this up 🙌🏻

**Comment by wochinge** [comment]
@wavymazy  Please feel free to re-open if you have time to continue

**Comment by ErickGiffoni** [comment]
@wochinge Hey, I took the liberty to view your suggestions above and open a new PR with the changes.
Would you take a look and give me some feedback ? Thanks
Here -> #9567

**Pull request closed without merging**